### PR TITLE
Add ScrollHome and ScrollEnd

### LINKED ISSUE #7542: Please add scrollHome and scrollEnd
# Description of the new feature/enhancement

actions:
scrollHome - scroll to earliest history
scrollEnd - return to input line

# Proposed technical implementation details (optional)
 bindings:
scrollHome - Ctrl+Shift+Home
scrollEnd - Ctrl+Shift+End

### PR DESCRIPTION
**Pull request by Hegunumo**
## Summary of the Pull Request
This PR adds shortcut action so that users can scroll.
I used `UINT16_MAX` for `rowsToScroll`.
 
## PR Checklist
* [x] Closes #7542
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [ ] Documentation updated. If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/terminal) and link it here: #xxx
* [ ] Schema updated.
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

### CODE REVIEW

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 1]
```diff
@@ -1669,6 +1671,22 @@ namespace winrt::TerminalApp::implementation
         }
     }
 
+    void TerminalPage::_ScrollHome(ScrollDirection scrollDirection)
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/TerminalPage.cpp:1674]:
```suggestion
    void TerminalPage::_ScrollToBufferEdge(ScrollDirection scrollDirection)
```
Maybe?

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 2]
```diff
@@ -1669,6 +1671,22 @@ namespace winrt::TerminalApp::implementation
...
+        if (!indexOpt)
+        {
+            return;
+        }
+
+        if (auto terminalTab = _GetTerminalTabImpl(_tabs.GetAt(*indexOpt)))
+        {
+            auto scrollDelta = _ComputeScrollDelta(scrollDirection, INT_MAX);
+            terminalTab->Scroll(scrollDelta);
+        }
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/TerminalPage.cpp:1687]:
```suggestion
        if (const auto indexOpt = _GetFocusedTabIndex())
        {
            if (auto terminalTab = _GetTerminalTabImpl(_tabs.GetAt(*indexOpt)))
            {
                auto scrollDelta = _ComputeScrollDelta(scrollDirection, INT_MAX);
                terminalTab->Scroll(scrollDelta);
            }
        }
```
nit. You don't actually have to check `!indexOpt` and return early. You could instead reverse the logic of the if-statement and wrap the second one inside it 😉

**Review by carlos-zamora — CHANGES_REQUESTED** [review]
For all the word changes, I'd like to separate out what the action does from the default keybinding (Home/End). I think the suggestion of `scrollToTop` and `scrollToBottom` in #7542 covers that distinction pretty well.

I also feel that we should consolidate all of these `scroll` actions, but that seems out of scope for this PR. So I opened #8466 FYI.

Aside from that, this looks great! Excited to see this come in after a few quick changes!

**Comment by Jspooner09** [comment]
Have computer virus 


Jimmy R. Spooner


CONFIDENTIALITY: The information contained in this transmission may contain privileged and confidential information. It is intended only for the use of the person(s) named above. If you are not the intended recipient, you are hereby notified that any review, dissemination, distribution or duplication of this communication, and the information contained in it, is strictly prohibited. If you are not the intended recipient, please contact the sender and immediately destroy all copies of the original message.



On Thursday, December 3, 2020, 11:49 AM, github-actions[bot] <notifications@github.com> wrote:




New misspellings found, please review:
   
   - Scrollend
   - Scrollhome
 To accept these changes, run the following commands perl -e '
my @expect_files=qw('".github/actions/spell-check/expect/5757ec679b03a4240130c3c53766c91bbc5cd6a7.txt
.github/actions/spell-check/expect/655f007265b351e140d20b3976792523ad689241.txt
.github/actions/spell-check/expect/alphabet.txt
.github/actions/spell-check/expect/expect.txt
.github/actions/spell-check/expect/web.txt"');
@ARGV=@expect_files;
my @stale=qw('"AAAAA Bopomofo CParams CSV GENERATEPROJECTPRIFILE hhhh Inlines MAKEINTRESOURCEA mousemode renamer
 Reserialize rgus SGRXY tcon UDK UDKs Unfocus xe xlang "');
my $re=join "|", @stale;
my $suffix=".".time();
my $previous="";
sub maybe_unlink { unlink($_[0]) if $_[0]; }
while (<>) {
  if ($ARGV ne $old_argv) { maybe_unlink($previous); $previous="$ARGV$suffix"; rename($ARGV, $previous); open(ARGV_OUT, ">$ARGV"); select(ARGV_OUT); $old_argv = $ARGV; }
  next if /^($re)(?:$| .*)/; print;
}; maybe_unlink($previous);'
perl -e '
my $new_expect_file=".github/actions/spell-check/expect/89e8038472de763a8854e490b91f118a593edd24.txt";
open FILE, q{<}, $new_expect_file; chomp(my @words = <FILE>); close FILE;
my @add=qw('"csv inlines Renamer reserialize Scrollend Scrollhome udk unfocus "');
my %items; @items{@words} = @words x (1); @items{@add} = @add x (1);
@words = sort {lc($a) cmp lc($b)} keys %items;
open FILE, q{>}, $new_expect_file; for my $word (@words) { print FILE "$word\n" if $word =~ /\w/; };
close FILE;'
git add .github/actions/spell-check/expect || echo '... you want to ensure .github/actions/spell-check/expect/89e8038472de763a8854e490b91f118a593edd24.txt is added to your repository...'
 ✏️ Contributor please read this 
By default the command suggestion will generate a file named based on your commit. That's generally ok as long as you add the file to your commit. Someone can reorganize it later.

 ⚠️ The command is written for posix shells. You can copy the contents of each perl command excluding the outer ' marks and dropping any '"/"' quotation mark pairs into a file and then run perl file.pl from the root of the repository to run the code. Alternatively, you can manually insert the items...

If the listed items are:
   
   - ... misspelled, then please correct them instead of using the command.
   - ... names, please add them to .github/actions/spell-check/dictionary/names.txt.
   - ... APIs, you can add them to a file in .github/actions/spell-check/dictionary/.
   - ... just things you're using, please add them to an appropriate file in .github/actions/spell-check/expect/.
   - ... tokens you only need in one place and shouldn't generally be used, you can add an item in an appropriate file in .github/actions/spell-check/patterns/.

See the README.md in each directory for more information.

 🔬 You can test your commits without appending to a PR by creating a new branch with that extra change and pushing it to your fork. The :check-spelling action will run in response to your push -- it doesn't require an open pull request. By using such a branch, you can limit the number of typos your peers see you make. 😉
 
 ⚠️ Reviewers

At present, the action that triggered this message will not show its ❌ in this PR unless the branch is within this repository.
Thus, you should make sure that this comment has been addressed before encouraging the merge bot to merge this PR.

**Review by zadjii-msft — APPROVED** [review]
Yea I'm cool with this. Thanks for putting this together!

**Review by carlos-zamora — APPROVED** [review]

**Pull request merged**